File: versus/objects/objects.c
#include <stdio.h>
#include <stdlib.h>

typedef struct {
    long x;
    long y;
} Point;

int main() {
    long total = 0;
    for (int i = 0; i < 1000000; i++) {
        Point *p = malloc(sizeof(Point));
        p->x = i;
        p->y = i + 1;
        total += p->x + p->y;
        free(p);
    }
    printf("%ld\n", total);
    return 0;
}

File: versus/objects/objects.rs
struct Point {
    x: i64,
    y: i64,
}

fn main() {
    let mut total: i64 = 0;
    for i in 0..1000000_i64 {
        let p = Box::new(Point { x: i, y: i + 1 });
        total += p.x + p.y;
    }
    println!("{}", total);
}
